Add typescript client page (#239)
ClouDesire/docusaurus: docs/api-typescript.md

---
id: api-typescript
title: TypeScript Client for Cloudesire API
sidebar_label: TypeScript Client
---

> You can find a TypeScript client on our [NPM Repository](https://nexus.cloudesire.com/#browse/browse:cloudesire-npm-public).

Reviewer: ```suggestion
You can find the TypeScript clients on our NPM Repository [https://nexus.cloudesire.com/repository/cloudesire-npm-public/](https://nexus.cloudesire.com/repository/cloudesire-npm-public/).
```

Author: da sloggato va usato l'URL con `#browse`

inoltre mi piace l'uso degli hyperlink

Reviewer: ma non è quello il repository npm, poi si chiedono quale url usare per il repository
l'url per browsarlo te lo dice se cerchi di andare sul vero url del repo npm

Author: capito, ho lasciato l'hyperlink alla versione browsabile e ho esplicitato l'URL del repository npm

> You can find a TypeScript client on our [Nexus Repository](https://nexus.cloudesire.com/#browse/browse:cloudesire-npm-public).

Reviewer: ```suggestion
You can find a TypeScript client on our [NPM Repository](https://nexus.cloudesire.com/#browse/browse:cloudesire-npm-public).
```

> Requires [axios HTTP client](https://github.com/axios/axios)

## Installation

> Install required packages, for example

Reviewer: ```suggestion
Install the required packages, for example:
```

```sh
npm install @cloudesire/backend-users-api @cloudesire/backend-billing-api
```

## Usage

```typescript
import axios, { AxiosResponse } from 'axios';
import { Configuration, LoginApi } from '@cloudesire/backend-users-api';
import { SubscriptionApi, SubscriptionDTO } from '@cloudesire/backend-billing-api';

> const url = '<cmw_api_url>';

Reviewer: ```suggestion
const url = '<cloudesire_api_url>';
```
